eZ/Publish/Core/Persistence/Legacy/Content/Type/Gateway/DoctrineDatabase.php

```diff
@@ -397,7 +397,7 @@ protected function setCommonTypeColumns(Query $q, ValueObject $type)
         )->set(
             $this->dbHandler->quoteColumn('language_mask'),
             $q->bindValue(
-                $this->languageMaskGenerator->generateLanguageMask($type->name),
+                $this->languageMaskGenerator->generateLanguageMask(array_flip($type->languageCodes)),
```

Comment:
Please add a new method `generateLanguageMaskFromLanguageCodes` next to that one in `MaskGenerator` instead of trying to reuse this one at all cost ;)

Out of scope note: we probably should tackle bulk loading of languages.